extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,175 @@
+    public String getName() throws ParsingException {
+        final String name = playlistData.getString("title");
+        if (name == null) {
+            throw new ParsingException("Could not get playlist name");
+        }
+        return name;
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            return getThumbnailUrlFromId(playlistId);
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist thumbnail", e);
+        }
+    }
+
+    @Override
+    public String getBannerUrl() {
+        return "";
+    }
+
+    @Override
+    public String getUploaderUrl() {
+        //Youtube mix are auto-generated
+        return "";
+    }
+
+    @Override
+    public String getUploaderName() {
+        //Youtube mix are auto-generated by YouTube
+        return "YouTube";
+    }
+
+    @Override
+    public String getUploaderAvatarUrl() {
+        //Youtube mix are auto-generated by YouTube
+        return "https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/YouTube_full-color_icon_(2017).svg/320px-YouTube_full-color_icon_(2017).svg.png";
+    }
+
+    @Override
+    public long getStreamCount() {
+        // Auto-generated playlist always start with 25 videos and are endless
+        return ListExtractor.ITEM_COUNT_INFINITE;
+    }
+
+    @Nonnull
+    @Override
+    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ExtractionException {
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        collectStreamsFrom(collector, playlistData.getArray(CONTENTS));
+        return new InfoItemsPage<>(collector, getNextPageUrl());
+    }
+
+    @Override
+    public String getNextPageUrl() throws ExtractionException {
+        final JsonObject lastStream = ((JsonObject) playlistData.getArray(CONTENTS)
+            .get(playlistData.getArray(CONTENTS).size() - 1));
+        if (lastStream == null || lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER) == null) {
+            throw new ExtractionException("Could not extract next page url");
+        }
+        return "https://youtube.com" + lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER)
+            .getObject("navigationEndpoint").getObject("commandMetadata")
+            .getObject("webCommandMetadata").getString("url") + "&pbj=1";
+    }
+
+    @Override
+    public InfoItemsPage<StreamInfoItem> getPage(final String pageUrl)
+            throws ExtractionException, IOException {
+        if (pageUrl == null || pageUrl.isEmpty()) {
+            throw new ExtractionException(
+                new IllegalArgumentException("Page url is empty or null"));
+        }
+
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        final JsonArray ajaxJson = getJsonResponse(pageUrl, getExtractorLocalization());
+        playlistData =
```

Comment:
Why are you assigning `playlistData` again here? If you call `getInitialPage()` after that, it won't return the initial page.

Reply:
Because i needed the updated `playlistData` for when `getNextPageUrl()` gets called multiple times.
I'just add a `initialPlaylistData` variable to the class

Reply:
`getNextPageUrl()` should always be relative to the initial page: https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/ListExtractor.java#L30

Reply:
oh, I completly missed that. I'll fix it